vizro-core/src/vizro/models/_controls/parameter.py

```diff
@@ -104,7 +105,34 @@ def _set_selector_title(self):
             self.selector.title = ", ".join({target.rsplit(".")[-1] for target in self.targets})
 
     def _set_actions(self):
+        from vizro.models import Filter
+
         if not self.selector.actions:
+            page_dynamic_filters = [
+                filter
+                for filter in cast(
+                    Iterable[Filter], model_manager._get_models(Filter, page=model_manager._get_model_page(self))
+                )
+                if filter._dynamic
+            ]
+
+            existing_target_ids: set[ModelID] = {cast(ModelID, target.split(".")[0]) for target in self.targets}
+            additional_targets: set[ModelID] = set()
+
+            # Extend parameter targets with dynamic filters linked to the same figure.
+            # Also, include dynamic filter targets to ensure new filter options are correctly calculated
+            # and filter targets are updated when filter values change.
+            for figure in self.targets:
+                figure_id, figure_arg = figure.split(".", 1)
+                if figure_arg.startswith("data_frame"):
+                    for filter in page_dynamic_filters:
+                        if figure_id in filter.targets:
+                            additional_targets.add(cast(ModelID, filter.id))
+                            # Exclude existing targets defined with the "dot" notation to avoid duplicates
+                            additional_targets |= set(filter.targets) - existing_target_ids
+
+            self.targets.extend(list(additional_targets))
```

Comment:
I think this could be simplified a bit, something like this (untested). The order of `self.targets` won't be maintained any more since we convert to set but I don't think that would be a problem?

```suggestion
        targets: set[ModelID] = set(self.targets)
        
        # Extend parameter targets with dynamic filters linked to the same figure.
        # Also, include dynamic filter targets to ensure new filter options are correctly calculated
        # and filter targets are updated when filter values change.
        for figure in self.targets:
            figure_id, figure_arg = figure.split(".", 1)
            if figure_arg.startswith("data_frame"):
                for filter in page_dynamic_filters:
                    if figure_id in filter.targets:
                        targets.add(cast(ModelID, filter.id))
                        targets |= set(filter.targets)
```
And then pass `targets=list(targets)` to `_parameter`.

Reply:
Decision: Let's keep duplicate targets like `["graph.x", "graph"]`.